CoverageReportIntegrationTest::test_coverage_report is failing

From e.g. https://github.com/google/oss-fuzz/runs/5309713893?check_suite_focus=true

```
=================================== FAILURES ===================================
______________ CoverageReportIntegrationTest.test_coverage_report ______________
[gw0] linux -- Python 3.8.12 /opt/hostedtoolcache/Python/3.8.12/x64/bin/python

self = <run_fuzzers_test.CoverageReportIntegrationTest testMethod=test_coverage_report>
_ = <MagicMock name='_upload_artifact_with_upload_js' id='140482468299728'>

    @mock.patch('filestore.github_actions._upload_artifact_with_upload_js')
    def test_coverage_report(self, _):
      """Tests generation of coverage reports end-to-end, from building to
      generation."""
    
      with test_helpers.docker_temp_dir() as temp_dir:
        shared = os.path.join(temp_dir, 'shared')
        os.mkdir(shared)
        copy_command = ('cp -r /opt/code_coverage /shared && '
                        'cp $(which llvm-profdata) /shared && '
                        'cp $(which llvm-cov) /shared')
        assert helper.docker_run([
            '-v', f'{shared}:/shared', 'gcr.io/oss-fuzz-base/base-runner', 'bash',
            '-c', copy_command
        ])
    
        os.environ['CODE_COVERAGE_SRC'] = os.path.join(shared, 'code_coverage')
        os.environ['PATH'] += os.pathsep + shared
        # Do coverage build.
        build_config = test_helpers.create_build_config(
            oss_fuzz_project_name=EXAMPLE_PROJECT,
            project_repo_name='oss-fuzz',
            workspace=temp_dir,
            git_sha='0b95fe1039ed7c38fea1f97078316bfc1030c[523](https://github.com/google/oss-fuzz/runs/5309713893?check_suite_focus=true#step:9:523)',
            base_commit='da07464[524](https://github.com/google/oss-fuzz/runs/5309713893?check_suite_focus=true#step:9:524)33dc18bae699e355a9821285d863c8',
            sanitizer=self.SANITIZER,
            cfl_platform='github',
            # Needed for test not to fail because of permissions issues.
            bad_build_check=False)
        self.assertTrue(build_fuzzers.build_fuzzers(build_config))
    
        # TODO(metzman): Get rid of this here and make 'compile' do this.
        chmod_command = ('chmod -R +r /out && '
                         'find /out -type d -exec chmod +x {} +')
    
        assert helper.docker_run([
            '-v', f'{os.path.join(temp_dir, "build-out")}:/out',
            'gcr.io/oss-fuzz-base/base-builder', 'bash', '-c', chmod_command
        ])
    
        # Generate report.
        run_config = test_helpers.create_run_config(fuzz_seconds=FUZZ_SECONDS,
                                                    workspace=temp_dir,
                                                    sanitizer=self.SANITIZER,
                                                    mode='coverage',
                                                    cfl_platform='github')
        result = run_fuzzers.run_fuzzers(run_config)
        self.assertEqual(result, run_fuzzers.RunFuzzersResult.NO_BUG_FOUND)
        expected_summary_path = os.path.join(
            TEST_DATA_PATH, 'example_coverage_report_summary.json')
        with open(expected_summary_path) as file_handle:
          expected_summary = json.loads(file_handle.read())
          actual_summary_path = os.path.join(temp_dir, 'cifuzz-coverage',
                                             'report', 'linux', 'summary.json')
>       with open(actual_summary_path) as file_handle:
E       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpfqnfmle8/cifuzz-coverage/report/linux/summary.json'
```

Related error:
```
------------------------------ Captured log call -------------------------------
ERROR    root:github_api.py:75 Request to https://api.github.com/repos/None/None/actions/artifacts?per_page=100&page=1 failed. Code: 401. Response: {'message': 'Bad credentials', 'documentation_url': 'https://docs.github.com/rest'}
ERROR    root:clusterfuzz_deployment.py:143 Failed to download corpus for target: do_stuff_fuzzer. Error: Github API request failed.
=========================== short test summary info ============================
FAILED infra/cifuzz/run_fuzzers_test.py::CoverageReportIntegrationTest::test_coverage_report```

Setting the correct [ACTIONS_RUNTIME_TOKEN](https://github.com/actions/toolkit/blob/fcb8c4ca798bd0d81a9b07e2f7dffee8397c4707/packages/artifact/src/internal/config-variables.ts#L34) (which is undocumented) instead gets us past the initial error, but now we see:

```
        expected_summary_path = os.path.join(
            TEST_DATA_PATH, 'example_coverage_report_summary.json')
        with open(expected_summary_path) as file_handle:
          expected_summary = json.loads(file_handle.read())
          actual_summary_path = os.path.join(temp_dir, 'cifuzz-coverage',
                                             'report', 'linux', 'summary.json')
>       with open(actual_summary_path) as file_handle:
E       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp2mypsq2s/cifuzz-coverage/report/linux/summary.json'
```

--- a/infra/cifuzz/filestore/github_actions/__init__.py
+++ b/infra/cifuzz/filestore/github_actions/__init__.py
@@ -63,7 +63,7 @@ class GithubActionsFilestore(filestore.BaseFilestore):
 
   def __init__(self, config):
     super().__init__(config)
-    self.github_api_http_headers = github_api.get_http_auth_headers(config)
+    self.github_api_http_headers = github_api.get_http_auth_headers()
 
   def _get_artifact_name(self, name):
     """Returns |name| prefixed with |self.ARITFACT_PREFIX| if it isn't already

--- a/infra/cifuzz/filestore/github_actions/github_actions_test.py
+++ b/infra/cifuzz/filestore/github_actions/github_actions_test.py
@@ -39,21 +39,20 @@ class GithubActionsFilestoreTest(fake_filesystem_unittest.TestCase):
   @mock.patch('platform_config.github._get_event_data', return_value={})
   def setUp(self, _):  # pylint: disable=arguments-differ
     test_helpers.patch_environ(self)
-    self.token = 'example githubtoken'
     self.owner = 'exampleowner'
     self.repo = 'examplerepo'
     os.environ['GITHUB_REPOSITORY'] = f'{self.owner}/{self.repo}'
     os.environ['GITHUB_EVENT_PATH'] = '/fake'
     os.environ['CFL_PLATFORM'] = 'github'
     os.environ['GITHUB_WORKSPACE'] = '/workspace'
-    self.config = test_helpers.create_run_config(token=self.token)
+    os.environ['ACTIONS_RUNTIME_TOKEN'] = 'githubtoken'
+    self.config = test_helpers.create_run_config()
     self.local_dir = '/local-dir'
     self.testcase = os.path.join(self.local_dir, 'testcase')
 
   def _get_expected_http_headers(self):
     return {
-        'Authorization': f'token {self.token}',
-        'Accept': 'application/vnd.github.v3+json',
+        'Authorization': 'Bearer githubtoken',
     }
 
   @mock.patch('filestore.github_actions.github_api.list_artifacts')

--- a/infra/cifuzz/filestore/github_actions/github_api.py
+++ b/infra/cifuzz/filestore/github_actions/github_api.py
@@ -35,12 +35,16 @@ _GET_ATTEMPTS = 3
 _GET_BACKOFF = 1
 
 
-def get_http_auth_headers(config):
+def get_http_auth_headers():
   """Returns HTTP headers for authentication to the API."""
-  authorization = f'token {config.token}'
+  # Undocumented token used for artifacts auth.
+  token = os.environ.get('ACTIONS_RUNTIME_TOKEN')
+  if not token:
+    return {}
+
+  authorization = f'Bearer {token}'
   return {
       'Authorization': authorization,
-      'Accept': 'application/vnd.github.v3+json'
   }
 
 

--- a/infra/cifuzz/filestore/github_actions/github_api_test.py
+++ b/infra/cifuzz/filestore/github_actions/github_api_test.py
@@ -31,11 +31,9 @@ class GetHttpAuthHeaders(unittest.TestCase):
 
   def test_get_http_auth_headers(self):
     """Tests that get_http_auth_headers returns the correct result."""
-    token = 'example githubtoken'
-    run_config = test_helpers.create_run_config(token=token)
+    test_helpers.patch_environ(self)
+    os.environ['ACTIONS_RUNTIME_TOKEN'] = 'githubtoken'
     expected_headers = {
-        'Authorization': f'token {token}',
-        'Accept': 'application/vnd.github.v3+json',
+        'Authorization': 'Bearer githubtoken',
     }
-    self.assertEqual(expected_headers,
-                     github_api.get_http_auth_headers(run_config))
+    self.assertEqual(expected_headers, github_api.get_http_auth_headers())

